addons/mail/models/discuss/discuss_channel.py

```diff
@@ -677,6 +678,8 @@ def _get_allowed_message_post_params(self):
 
     @api.returns('mail.message', lambda value: value.id)
     def message_post(self, *, message_type='notification', **kwargs):
+        if self.read_only:
```

Comment:
Not necessary as it should crash due to the correct implementation of access rights check on message model. We don't need a UserError if the flow is already raising at controller level, an AccessError raised by classic access rights check is clear enough at this point.